soh/soh/Enhancements/mods.cpp

```diff
@@ -456,70 +456,93 @@ void RegisterDaytimeGoldSkultullas() {
     });
 }
 
-void RegisterHyperBosses() {
-    GameInteractor::Instance->RegisterGameHook<GameInteractor::OnActorUpdate>([](void* refActor) {
-        // Run the update function a second time to make bosses move and act twice as fast.
+bool IsHyperBossesActive() {
+    return CVarGetInteger("gHyperBosses", 0) ||
+           (IS_BOSS_RUSH && gSaveContext.bossRushOptions[BR_OPTIONS_HYPERBOSSES] == BR_CHOICE_HYPERBOSSES_YES);
+}
 
-        Player* player = GET_PLAYER(gPlayState);
-        Actor* actor = static_cast<Actor*>(refActor);
+void UpdateHyperBossesState() {
+    static uint32_t actorUpdateHookId = 0;
+    if (actorUpdateHookId != 0) {
+        GameInteractor::Instance->UnregisterGameHook<GameInteractor::OnActorUpdate>(actorUpdateHookId);
```

Comment:
Unless I'm missing something, this will unregister and register the hook every frame while it's enabled which was probably not the intent.
I think adding a `!IsHyperBossesActive()` check would fix it.

Reply:
Not quite, the `UpdateHyperBossesState` will be run in the three following scenarios:
- On boot (when the `RegisterHyperBosses` function is called)
- When a game is loaded (this is primarily to account for bosh rush mode saves)
- When the checkbox is clicked in the menubar

In those three scenarios you are correct, it will unregister, then re-register a hook _if_ applicable.

Reply:
Makes sense, completely missed that this isn't called every frame.